### PR TITLE
Update k8s.io/kubernetes to fix GO-2024-2746

### PR DESCRIPTION
**Pull request by hime**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:
Update k8s.io/kubernetes to fix GO-2024-2746

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
Update k8s.io/kubernetes to fix GO-2024-2746
```

### CODE REVIEW

**Comment by tyuchn** [comment]
/ok-to-test
Kindly reminder, don't forget to add a release note in this PR

File: pkg/releaselock/rpc.go [hunk 1]
```diff
@@ -118,7 +118,7 @@ func ReleaseLock(hostIP, clientIP string) error {
 		ServiceMethod: inbandLockReleaseProcedureName,
 		Seq:           uint64(time.Now().UnixNano()),
 	}
-	klog.Infof("Sending RPC request %+v from GKE node IP to Filestore IP %s", request, clientIP, hostIP)
```
Comment (tyuchn) [inline pkg/releaselock/rpc.go:121]: I think this should change to 
```
klog.Infof("Sending RPC request %+v from GKE node IP %s to Filestore IP %s", request, clientIP, hostIP)
```

**Review by tyuchn — CHANGES_REQUESTED** [review]
/lgtm

File: pkg/releaselock/rpc.go (hunk 1 above)
Comment (hime) [reply pkg/releaselock/rpc.go:121]: Done.

**Review by tyuchn — APPROVED** [review]

**Pull request merged**